browser-extension/mv2/src/modules/EventActions.js

```diff
@@ -1,74 +1,109 @@
+/* SCOPE */
 const EventActions = {};
+/* STATE */
+EventActions.eventsToWrite = [];
+EventActions.eventWriterInterval = null;
+EventActions.batchesWaitingForAck = [];
+EventActions.STORE_EVENTS_KEY = "eventBatches";
 
-let eventsToWrite = [];
-let eventWriterInterval = null;
-
-const LOCAL_ANALYTICS_KEY = "analytics_event";
+/* UTILITIES */
 
 EventActions.queueEventToWrite = (event) => {
-  // todo: generate uuid for each event
-  eventsToWrite.push(event);
+  EventActions.eventsToWrite.push(event);
 };
 
-EventActions.getEvents = async () => {
-  const events = RQ.StorageService.getRecord(LOCAL_ANALYTICS_KEY)
-    .then((storedEvents) => {
-      // these are considered the events that are ready to be sent
-      // not considering events in `eventsBatch` here
-      return storedEvents;
+EventActions.getEventBatches = async () => {
+  const eventBatches = RQ.StorageService.getRecord(
```

Comment:
```suggestion
  return RQ.StorageService.getRecord(
```
the variable confuses if its a promise. Lets return it straightaway